tensorflow_datasets/core/dataset_builder.py

```diff
@@ -1071,22 +1126,31 @@ def _generate_examples(self, **kwargs):
     """
     raise NotImplementedError()
 
-  def _download_and_prepare(self, dl_manager, download_config):
+  # pylint: disable=arguments-differ
+  def _download_and_prepare(
+      self,
+      dl_manager: download.DownloadManager,
+      download_config: download.DownloadConfig) -> None:
     # Extract max_examples_per_split and forward it to _prepare_split
     super(GeneratorBasedBuilder, self)._download_and_prepare(
         dl_manager=dl_manager,
         max_examples_per_split=download_config.max_examples_per_split,
     )
 
-  def _prepare_split_legacy(self, generator, split_info):
+  def _prepare_split_legacy(self,
+                            generator: Iterable,
```

Comment:
`Iterator[X]`

Reply:
I thin it is `tuple` so changed to `Iterable[tf.tuple]`